core/src/main/java/org/apache/iceberg/MetricsConfig.java

```diff
@@ -94,15 +96,21 @@ public static MetricsConfig getDefault() {
    **/
   @Deprecated
   public static MetricsConfig fromProperties(Map<String, String> props) {
-    return from(props, null);
+    return from(props, null, DEFAULT_WRITE_METRICS_MODE_DEFAULT);
   }
 
   /**
    * Creates a metrics config from a table.
    * @param table iceberg table
    */
   public static MetricsConfig forTable(Table table) {
-    return from(table.properties(), table.sortOrder());
+    String defaultMode;
+    if (table.schema().columns().size() <= MAX_COLUMNS) {
```

Comment:
Is there a way to override this behavior?

Reply:
Good point, I can make a table config , does that sound reasonable?

Reply:
"defaultMetricCollection : all" or something? Don't we already have that?

Also this feels like it should probably be a catalog level prop :shrug:

Reply:
Was initially thinking like 'write.metadata.metrics.max.columns'='100'?  Yea good point probably catalog level prop is cleaner

Reply:
Took another look, and actually it's a bit messy to make this a catalog property in the current code (the Table is serialized and sent to writers, but Catalog is not..).  

Was thinking it makes sense as a table property as well (ie, if i know this table is worth optimizing, set the max columns to be higher).  May be a bit awkward to have a different catalog just for this.  And if user really wants to have one global setting, this change is coming: https://github.com/apache/iceberg/pull/4011

Reply:
I think we may already have what we need to do this. Right now, the default metrics mode is set by `write.metadata.metrics.default`. I'd propose the following:
* If `write.metadata.metrics.default` explicitly set in table properties, always use it
* If `write.metadata.metrics.default` is not set, check the number of columns
    * If `numCols < 32` then use `truncate[16]`, the current default
    * If `numCols >= 32` then use `none`

That seems like a reasonable way to make this customizable.

Reply:
I think this is okay for now. Tables will still respect whatever is set as `write.metadata.metrics.default` so this really just changes Iceberg's default in a reasonable way. It is also good to note that metrics for sort columns are automatically promoted to at least `truncate[16]` so it isn't as though we're losing _all_ stats.